src/Utility/Utils.cs

```diff
@@ -52,7 +59,75 @@ internal static PowerShell NewPwshInstance()
                 }
             }
 
-            return PowerShell.Create(s_iss);
+            var pwsh = PowerShell.Create(s_iss);
+            if (s_globalVariables == null)
+            {
+                // Get the names of the built-in global variables
+                ICollection<PSVariable> globalVars = GetGlobalVariables(pwsh);
+                s_globalVariables = new HashSet<string>(globalVars.Count + 3, StringComparer.OrdinalIgnoreCase)
```

Comment:
`PSCmdlet` doesn't show up in the global scope after evaluating a regular 'profile.ps1' that doesn't have a `param` block with `CmdletBinding()` attribute, so I don't put it here.
Actually, all other automatic variables that created along with invoking a script will be handled by the `GetType() != typeof(PSVaraible)` check below, because they will be of the type `LoaclVariable`.

Reply:
I see, gotcha!